### PR TITLE
feat: make log rotation settings configurable

### PR DESCRIPTION
**Pull request by embroede**
Make log rotation MaxSize, MaxAge, and MaxBackups configurable.

## Description

Enable configuration of some log rotation settings.

## Related Issue

https://github.com/dragonflyoss/Dragonfly2/issues/2956

## Motivation and Context

The current hard-coded settings can lead to up to 20GB of disk space used per log file. It is reasonable to expect that some users may want to enable verbose mode, but still not allow so much space to be used.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by embroede** [comment]
I'm seeing segfaults in some of these tests, which don't seem related to the changes:
```
kubectl command: "kubectl" "-n" "dragonfly-system" "exec" "-c" "dfdaemon" "dragonfly-dfdaemon-7xpts" "--" "/opt/dragonfly/bin/dfget" "--disable-back-source" "-O" "/tmp/d7y.out" "-H" "Range: bytes=1-17" "http://file-server.dragonfly-e2e.svc/kind/etc/fstab"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x90 pc=0x149ce52]
```

**Comment by embroede** [comment]
> I'm seeing segfaults in some of these tests, which don't seem related to the changes:
> 
> ```
> kubectl command: "kubectl" "-n" "dragonfly-system" "exec" "-c" "dfdaemon" "dragonfly-dfdaemon-7xpts" "--" "/opt/dragonfly/bin/dfget" "--disable-back-source" "-O" "/tmp/d7y.out" "-H" "Range: bytes=1-17" "http://file-server.dragonfly-e2e.svc/kind/etc/fstab"
> panic: runtime error: invalid memory address or nil pointer dereference
> [signal SIGSEGV: segmentation violation code=0x1 addr=0x90 pc=0x149ce52]
> ```

Sorry that's wrong, these are related, I'll get these fixed.

File: internal/dflog/logcore.go [hunk 1]
```diff
@@ -52,12 +58,28 @@ var customCoreLevel atomic.Bool
 var grpcLevel = zap.NewAtomicLevelAt(zapcore.WarnLevel)
 var customGrpcLevel atomic.Bool
 
-func CreateLogger(filePath string, compress bool, stats bool, verbose bool) (*zap.Logger, zap.AtomicLevel, error) {
+func CreateLogger(filePath string, compress bool, stats bool, verbose bool, config LogRotateConfig) (*zap.Logger, zap.AtomicLevel, error) {
+
+	maxSize := defaultRotateMaxSize
```
Comment (gaius-qi) [inline internal/dflog/logcore.go:63]: Set default value in each service's config. 
Scheduler: https://github.com/dragonflyoss/Dragonfly2/blob/main/scheduler/config/constants.go.
Manager: https://github.com/dragonflyoss/Dragonfly2/blob/main/manager/config/constants.go.
Client: https://github.com/dragonflyoss/Dragonfly2/blob/main/client/config/constants.go.

**Review by gaius-qi — CHANGES_REQUESTED** [review]

File: internal/dflog/logcore.go (hunk 1 above)
Comment (embroede) [reply internal/dflog/logcore.go]: Sure, this is done now. One question, I noticed `logger.InitDaemon()` is called from `SetupDaemon()` in `log.go`. But I cannot see `SetupDaemon()` being called anywhere within `Dragonfly2`. What is the purpose of `SetupDaemon()`?

**Review by gaius-qi — CHANGES_REQUESTED** [review]
Please add rotate params to `SetupDaemon`. No more questions.

File: internal/dflog/logcore.go (hunk 1 above)
Comment (gaius-qi) [reply internal/dflog/logcore.go]: If there are other services that want to use dragonfly's log, they will be initialized through this `SetupDaemon`.

**Review by gaius-qi — APPROVED** [review]
LGTM

**Review by jim3ma — APPROVED** [review]
LGTM

File: internal/dflog/logcore.go (hunk 1 above)
Comment (jim3ma) [reply internal/dflog/logcore.go]: The `SetupDaemon` is deprecated, we will clean up it, resolved here.

**Pull request merged**